lib/sequent/support/database.rb

```diff
@@ -30,17 +31,51 @@ def self.read_config(env)
       end
 
       def self.create!(db_config)
-        ActiveRecord::Base.establish_connection(db_config.merge(database: 'postgres'))
-        ActiveRecord::Base.connection.create_database(db_config[:database])
+        if Sequent.configuration.can_use_multiple_databases?
+          ActiveRecord.legacy_connection_handling = false
+          ActiveRecord::Base.configurations = db_config.deep_merge(
+            Sequent.configuration.primary_database_key => {database: 'postgres'},
+          ).stringify_keys
+
+          ActiveRecord::Base.connects_to database: {
+            Sequent.configuration.primary_database_role => Sequent.configuration.primary_database_key,
+          }
+          ActiveRecord::Base
+            .connection
+            .create_database(db_config[Sequent.configuration.primary_database_key][:database])
+        else
+          ActiveRecord::Base.establish_connection(db_config.merge(database: 'postgres'))
+          ActiveRecord::Base.connection.create_database(db_config[:database])
+        end
       end
 
       def self.drop!(db_config)
-        ActiveRecord::Base.establish_connection(db_config.merge(database: 'postgres'))
-        ActiveRecord::Base.connection.drop_database(db_config[:database])
+        if Sequent.configuration.can_use_multiple_databases?
```

Comment:
This block of code looks similar to that in `create`. Can be dry it up into a separate method/class?